aptos-move/framework/aptos-token-objects/sources/token.move

```diff
@@ -254,6 +281,21 @@ module aptos_token_objects::token {
         }
     }
 
+    /// This method allows minting/burning to happen in parallel, making it efficient.
+    public fun index_snapshot<T: key>(token: &Object<T>): AggregatorSnapshot<u64> acquires Token, TokenAppendix1 {
+        let token_address = object::object_address(token);
+        if (exists<TokenAppendix1>(token_address)) {
+            aggregator_v2::copy_snapshot(&borrow_global<TokenAppendix1>(token_address).index)
+        } else {
+            aggregator_v2::create_snapshot(borrow(token).deprecated_index)
+        }
+    }
+
+    #[view]
+    public fun index_view<T: key>(token: Object<T>): u64 acquires Token, TokenAppendix1 {
```

Comment:
Who calls `index_view` function? The indexer?

If we can combine `index_view` and `index_snapshot` methods, then we don't need to have the new native function `copy_snapshot` right? 
In fact, can we simply use the copy capability on AggregatorSnapshots instead of `copy_snapshot` native function?